space-cli/actions.go

```diff
@@ -26,43 +32,71 @@ func setLogLevel(loglevel string) {
 	}
 }
 
-func actionStartCode(c *cli.Context) error {
-	service, loginResp, err := cmd.CodeStart()
+func actionApply(c *cli.Context) error {
+	args := os.Args
+	if len(args) != 3 {
+		return fmt.Errorf("incorrect number of arguments provided")
+	}
+	data, err := ioutil.ReadFile(args[2])
 	if err != nil {
 		return err
 	}
-	actionCodeStruct := &model.ActionCode{
-		Service:  service,
-		IsDeploy: true, //
-	}
-	if err := cmd.RunDockerFile(actionCodeStruct, loginResp); err != nil {
+	fileContent := new(model.GitOp)
+	if err := json.Unmarshal(data, fileContent); err != nil {
 		return err
 	}
-	return nil
-}
+	if fileContent.Type != "service" {
+		return fmt.Errorf("invalid type found in serivce file type should be (service)")
+	}
+	service := fileContent.Spec.(*model.Service)
+	service.ID = fileContent.Meta["id"]
+	service.ProjectID = fileContent.Meta["projectId"]
+	service.Version = fileContent.Meta["version"]
 
-func actionBuildCode(c *cli.Context) error {
-	service, loginResp, err := cmd.CodeStart()
+	requestBody, err := json.Marshal(&fileContent.Spec)
 	if err != nil {
+		logrus.Error("error in apply service unable to marshal data - %v", err)
 		return err
 	}
-	actionCodeStruct := &model.ActionCode{
-		Service:  service,
-		IsDeploy: false,
-	}
-	if err := cmd.RunDockerFile(actionCodeStruct, loginResp); err != nil {
+	urlPath := strings.Replace(fileContent.Api, "{projectId}", service.ProjectID, 1)
+	resp, err := http.Post(fmt.Sprintf("http://localhost:4122%s", urlPath), "application/json", bytes.NewBuffer(requestBody))
```

Comment:
How did you get this url??. I would have taken it from server url field in the selected account